### PR TITLE
[feature] Allow passing GraphQLSchema directly

### PR DESCRIPTION
**Pull request by zth**
This allows passing an already instantiated `GraphQLSchema` in addition to an introspection fetcher or an actual introspection result. Use case is I've got somewhere where I already have the instantiated schema but not access to the introspection result (and not the endpoint to get it either).

Please let me know if there are any questions or anything I can clarify further.

### CODE REVIEW

**Comment by IvanGoncharov** [comment]
@zth Thanks for PR 👍 
After thinking about this, the ideal long term solution would be to copy the interface of `GraphiQL` and add separate `schema` and `fetcher` properties.

Do you have time to work on such PR?

If not you can use `introspectionFromSchema` function as temporary solution, see here:
https://github.com/APIs-guru/graphql-voyager/blob/02ce607e49ad9169c5af58413b35e9b81300e80c/demo/IntrospectionModal.tsx#L77

**Comment by zth** [comment]
@IvanGoncharov thanks for having a look! Sounds like a good approach. I pushed some more changes, is that more in line with what you had in mind?

**Comment by IvanGoncharov** [comment]
>  I pushed some more changes, is that more in line with what you had in mind?

@zth Thanks a lot for the changes 👍 
Since we currently in the RC phase we can do breaking changes so we can literally copy the GraphiQL interface:
* rename `introspection` to `fetcher` and restrict it to accept only functions with the same signature as in GraphiQL
* first we should use `schema` and only if it absent use `fetcher` to fetch introspection

**Comment by zth** [comment]
Ok, great! I'll work on this as soon as I have a chance. Talk to you in a while!

**Comment by zth** [comment]
@IvanGoncharov there, I've pushed some new changes. Please have a look and get back to me.

File: README.md [hunk 1]
```diff
@@ -29,7 +29,9 @@ module system it is exported as `GraphQLVoyager` global variable.
 ### Properties
 `Voyager` component accepts the following properties:
 
-+ `introspection` [`object` or function: `(query: string) => Promise`] - the server introspection response. If `function` is provided GraphQL Voyager will pass introspection query as a first function parameter. Function should return `Promise` which resolves to introspection response object.
++ `fetcher` _(optional)_ [function: `(query: string) => Promise`] - a function for fetching an introspection of your schema. GraphQL Voyager will pass an introspection query as a first function parameter. Function should return `Promise` which resolves to introspection response object.
```
Comment (IvanGoncharov) [inline README.md]: GraphiQL pass an object with `query` and `operationName` (also possible `variables`) so you send it directly to the server:
https://github.com/graphql/graphiql/blob/master/packages/graphiql/src/components/GraphiQL.js#L580

Comment (IvanGoncharov) [reply README.md]: We should do the same so you can use same function for both Voyager and GraphiQL

File: demo/index.tsx [hunk 2]
```diff
@@ -45,7 +45,7 @@ export default class Demo extends React.Component {
 
     return (
       <MuiThemeProvider theme={theme}>
-        <GraphQLVoyager introspection={introspection}>
+        <GraphQLVoyager fetcher={introspection}>
```
Comment (IvanGoncharov) [inline demo/index.tsx]: `introspection` can be an JSON object here.
Can you please test it with `SDL` and `Introspection` tabs?

**Comment by zth** [comment]
Just a quick status update: I've been sidetracked and currently won't have time to continue this, but I'll ping you again as soon as I do! Thanks for reviewing :+1: :smile:

**Comment by IvanGoncharov** [comment]
Implemented in #340

**Pull request closed without merging**